client/src/routes/modals/CompletedSessionsModal/CompletedSessionsModal.tsx

```diff
@@ -0,0 +1,226 @@
+    selectedMode,
+  ]);
+
+  const Footer = useMemo(
+    () => (
+      <>
+        <Spacer32 />
+        {filterSetting === 'feedback' && (
+          <Row>
+            <FilterStatus
+              selected={selectedFeedback === true}
+              Icon={ThumbsUpWithoutPadding}
+              onPress={() =>
+                setSelectedFeedback(selectedFeedback ? undefined : true)
+              }
+              heading={`${positiveFeedbacks.length}`}
+              description={t('meaninful')}
+            />
+            <Spacer16 />
+            <FilterStatus
+              selected={selectedFeedback === false}
+              Icon={ThumbsDownWithoutPadding}
+              onPress={() =>
+                setSelectedFeedback(
+                  selectedFeedback === false ? undefined : false,
+                )
+              }
+              heading={`${negativeFeedbacks.length}`}
+              description={t('notMeaninful')}
+            />
+          </Row>
+        )}
+
+        {filterSetting === 'mode' && (
+          <Row>
+            <FilterStatus
+              Icon={MeIcon}
+              selected={selectedMode === SessionMode.async}
+              onPress={() =>
+                setSelectedMode(
+                  selectedMode !== SessionMode.async
+                    ? SessionMode.async
+                    : undefined,
+                )
+              }
+              heading={`${asyncSessions?.length}`}
+              description={t('async')}
+            />
+            <Spacer16 />
+            {privateSessions?.length && (
+              <FilterStatus
+                Icon={FriendsIcon}
+                selected={selectedMode === SessionType.private}
+                onPress={() =>
+                  setSelectedMode(
+                    selectedMode !== SessionType.private
+                      ? SessionType.private
+                      : undefined,
+                  )
+                }
+                heading={`${privateSessions?.length}`}
+                description={t('private')}
+              />
+            )}
+            <Spacer16 />
+            {publicSessions?.length && (
+              <FilterStatus
+                Icon={CommunityIcon}
+                selected={selectedMode === SessionType.public}
+                onPress={() =>
+                  setSelectedMode(
+                    selectedMode !== SessionType.public
+                      ? SessionType.public
+                      : undefined,
+                  )
+                }
+                heading={`${publicSessions?.length}`}
+                description={t('public')}
+              />
+            )}
+          </Row>
+        )}
+        <Spacer32 />
+      </>
+    ),
+    [
+      positiveFeedbacks?.length,
+      negativeFeedbacks?.length,
+      asyncSessions?.length,
+      privateSessions?.length,
+      publicSessions?.length,
+      filterSetting,
+      selectedMode,
+      selectedFeedback,
+      t,
+    ],
+  );
```

Comment:
These situations are so tricky. Like needing contextual variables in something that should maybe be a separate component 🤔 Would be nice to separare some how. But I can't come up with a good solution from the top of my head. So do what you ant with this comment 😄 Will think more about it though!